openlibrary/plugins/openlibrary/api.py

```diff
@@ -708,3 +711,72 @@ def GET(self):
             img.save(buf, format='PNG')
             web.header("Content-Type", "image/png")
             return delegate.RawText(buf.getvalue())
+
+
+class bestbook_award(delegate.page):
+    path = r"/works/OL(\d+)W/awards"
+    encoding = "json"
+
+    @jsonapi
+    def POST(self, work_id):
+        """Store Bestbook award
+
+        Args:
+            work_id (int): unique id for each book
+        """
+        OPS = ["add", "remove", "update"]
+        i = web.input(op="add", edition_key=None, topic=None, comment="")
+
+        edition_id = i.edition_key and int(extract_numeric_id_from_olid(i.edition_key))
+        errors = []
+
+        if user := accounts.get_current_user():
+            try:
+                username = user.key.split('/')[2]
+                if i.op in ["add", "update"]:
+                    # Make sure the topic is free
+                    if i.op == "update":
+                        Bestbook.remove(username, topic=i.topic)
```

Comment:
Hmm could `i.topic` be `None`? Would that cause the `.remove` method to `ValueError` ?